credit card listing without actions(add, edit, delete)

jonathandelefortrie opened the pull request:
#### What?

Implementation of the payment methods listing.
* Credit cards group by payment providers
* Includes of the payment method icons
* Mobile and desktop layout
* Translations

#### Ticket

- [Link](https://jira.bigcommerce.com/browse/PAYMENTS-3168)

#### Screenshots

Desktop
<img width="1009" alt="screen shot 2018-09-04 at 2 54 11 pm" src="https://user-images.githubusercontent.com/2579218/45010772-731eb800-b052-11e8-9fa7-376859d877e3.png">

Mobile
<img width="377" alt="screen shot 2018-09-04 at 1 35 02 pm" src="https://user-images.githubusercontent.com/2579218/45008706-7b252a80-b047-11e8-8867-f8fff070a027.png">

Navigation Mobile
<img width="377" alt="screen shot 2018-09-04 at 1 35 23 pm" src="https://user-images.githubusercontent.com/2579218/45008716-8c6e3700-b047-11e8-909c-532e5457aa6b.png">

ping @bigcommerce/payments

jackosaurus commented on lang/en.json:
```diff
@@ -355,8 +355,21 @@
+                "diners": "Diners",
+                "discover": "Discover",
+                "mastercard": "Master Card",
+                "visa": "Visa"
```
Can you please also add
```
        'jcb' => 'jcb',
        'dankort' => 'dankort',
        'maestro' => 'maestro'
```

jackosaurus commented:
ping @bigcommerce/storefront-team for visibility

danieldelcore commented:
Nice work, Jonathan 🎉

jackosaurus commented on lang/en.json:
```diff
@@ -355,8 +355,21 @@
+            "card_ending_in": "ending in {last_four}",
+            "card_expiry": "{month}/{year}",
+            "new_payment_method": "Add new payment method",
+            "no_methods": "You currently have no payment methods added to your profile",
```
prefer "account" over "profile" to keep language in that area consistent

danieldelcore commented on templates/components/common/navigation-menu.html:
```diff
@@ -80,8 +80,13 @@
                             <a class="navPage-subMenu-action navPages-action" href="{{urls.account.inbox}}">{{lang 'account.mobile_nav.messages'}}</a>
                         </li>
                         <li class="navPage-subMenu-item">
-                            <a class="navPage-subMenu-action navPages-action" href="{{urls.account.addresses}}">{{lang 'account.nav.addresses'}}</span></a>
```
good catch 👍

danieldelcore commented on templates/components/account/payment-methods-list.html:
```diff
@@ -0,0 +1,46 @@
+                        {{#if brand '===' 'visa'}}
+                            <img class="paymentMethod-icon" src="{{cdn 'img/payment-methods/visa.svg'}}" alt="{{lang 'account.payment_methods.card_types.visa'}}" title="{{lang 'account.payment_methods.card_types.visa'}}">
+                            <span class="paymentMethod-brand">{{lang 'account.payment_methods.card_types.visa'}} {{lang 'account.payment_methods.card_ending_in' last_four=last_4}}</span>
+                        {{/if}}
```
We actually store other credit card types (ie brands) in our vaults. Would you be able to add a fallback strategy for cards that do not match any of the brands listed here? 

I would suggest a generic credit card logo or no logo and copy that reads something like `Credit card ending in XXXX`.

🍹

junedkazi commented on config.json:
```diff
@@ -30,7 +30,8 @@
       "google_amp",
-      "customized_checkout"
+      "customized_checkout",
+      "account_payment_methods"
```
This will need an entry in theme registry features list as well. See https://github.com/bigcommerce/theme-registry/blob/master/app/schemas/tag.json#L2 and here https://github.com/bigcommerce/stencil-cli/blob/e5c3e96e3a3d10e4423afec418c2c8f3686d2d02/lib/themeConfig.schema.json#L33 as well.

junedkazi replied:
You need this bcoz for marketplace theme we validate against a fixed set of values.

junedkazi commented on templates/components/account/payment-methods-list.html:
```diff
@@ -0,0 +1,50 @@
+                <div class="panel-body">
+                    <div class="paymentMethod-card">
+                        {{#if brand '===' 'american_express'}}
+                            <img class="paymentMethod-icon" src="{{cdn 'img/payment-methods/american-express.svg'}}" alt="{{lang 'account.payment_methods.card_types.american_express'}}" title="{{lang 'account.payment_methods.card_types.american_express'}}">
```
Can we simplify this `if .. else if .. else` structure. Can't we make the brand dynamic in the html.

junedkazi commented on templates/components/amp/common/navigation-menu.html:
```diff
@@ -70,8 +70,13 @@
                         </li>
                         <li class="navPage-subMenu-item">
-                            <a class="navPage-subMenu-action navPages-action" href="{{urls.account.addresses}}">{{lang 'account.nav.addresses'}}</span></a>
+                            <a class="navPage-subMenu-action navPages-action" href="{{urls.account.addresses}}">{{lang 'account.nav.addresses'}}</a>
```
good catch 👏

junedkazi commented on templates/components/common/navigation-menu.html:
```diff
@@ -80,8 +80,13 @@
                         </li>
                         <li class="navPage-subMenu-item">
-                            <a class="navPage-subMenu-action navPages-action" href="{{urls.account.addresses}}">{{lang 'account.nav.addresses'}}</span></a>
+                            <a class="navPage-subMenu-action navPages-action" href="{{urls.account.addresses}}">{{lang 'account.nav.addresses'}}</a>
```
👏

jonathandelefortrie commented:
Hey @junedkazi , I have simplified the logic in the template so it becomes more dynamic. I have added a list of supported icons in the ```config.json``` to ensure a valid display of the icon.  Please let me know if this a good approach and the right place for keeping those values.

jonathandelefortrie commented:
Hey @junedkazi , those changes will get merged into a feature branch as there will be more changes to come. Should I keep the update of change log when doing the PR against master?

junedkazi commented:
@jonathandelefortrie lets do one at the very end about when we add it to master.

jonathandelefortrie commented:
Great! Thanks for the review
